src/components/settings-row/settings-row.test.js

```diff
@@ -0,0 +1,91 @@
+import * as React from "react";
+import SettingsRow from "./settings-row.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { getDataElementByValue } from "../../../cypress/locators";
+import {
+  settingsRowPreview,
+  settingsRowChildren,
+  settingsRowDescription,
+} from "../../../cypress/locators/settings-row/index";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData1 = [
+  "https://carbon.sage.com",
+  "https://carbon.sage.com/?path=/docs/setting-row--default-story",
+];
+
+const SettingsRowComponent = ({ ...props }) => {
+  return <SettingsRow title="title" description="description" {...props} />;
+};
+
+context("Tests for SettingsRow component", () => {
+  describe("should check SettingsRow component properties", () => {
+    it.each(testData)(
+      "should check %s as title for SettingsRow component",
+      (title) => {
+        CypressMountWithProviders(<SettingsRowComponent title={title} />);
+        getDataElementByValue("title").should("have.text", title);
+      }
+    );
+
+    it.each(testData)(
+      "should check %s as description for SettingsRow component",
+      (description) => {
+        CypressMountWithProviders(
+          <SettingsRowComponent description={description} />
+        );
+        settingsRowDescription().should("have.text", description);
+      }
+    );
+
+    it.each(testData1)(
```

Comment:
I think this is meant to be a `Link` component, e.g. `<Link>This is a link</Link>`, rather than a URL type link as it states in Docs for the `description` prop "A string or JSX object...".